packages/extension/src/background/tokenMessaging.ts

```diff
@@ -0,0 +1,46 @@
+import { TokenMessage } from "../shared/messages/TokenMessage"
+import { defaultNetwork } from "../shared/network"
+import { hasToken } from "../shared/token/storage"
+import { HandleMessage, UnhandledMessage } from "./background"
+
+export const handleTokenMessaging: HandleMessage<TokenMessage> = async ({
+  msg,
+  background: { actionQueue, wallet },
+  respond,
+}) => {
+  switch (msg.type) {
+    case "REQUEST_TOKEN": {
+      const selectedAccount = await wallet.getSelectedAccount()
```

Comment:
The selected account might not be the network that is connected by the dApp. We probably need to pass by the network id as well?

Reply:
Right now `networkId` is optional in the message, I will make it take highest precedence and fallback to the network id of the account, WDTY?

Reply:
That's the approach we use in general I think, so good to me